### PR TITLE
Inline binops

### PR DESCRIPTION
**Pull request by R4PaSs**
This PR re-introduces several binary operations implemented as intern to avoid indirection through FFI in some specific cases to improve performance notably in `core/text/native`

Also a style change for `abs` which could easily be inlined and burned by eyes when I saw it, I can remove the commit if someone objects to the modification

### CODE REVIEW

**Comment by privat** [comment]
seems good. check why tests failed

**Comment by R4PaSs** [comment]
Just tests to update, nituml discovers new properties and lines changed in `kernel` for error tests, I'll send a reroll soon

File: lib/core/kernel.nit [hunk 1]
```diff
@@ -644,17 +644,23 @@ universal Byte
 	# `i` bits shift fo the left
 	#
 	#     assert 5u8 << 1    == 10u8
-	fun <<(i: Int): Byte `{ return self << i; `}
+	fun <<(i: Int): Byte is intern do return lsh(i)
+
+	private fun lsh(i: Int): Byte `{ return self << i; `}
```
Comment (xymus) [inline lib/core/kernel.nit]: Is this a fallback to support the interpreter? Just curious, could you have written the following?

~~~
fun <<(i: Int): Byte is intern `{ return self << i; `}
~~~

**Comment by xymus** [comment]
+1

File: lib/core/kernel.nit (hunk 1 above)
Comment (R4PaSs) [inline lib/core/kernel.nit]: Yup, that's a fallback for any engine which does not support this implementation internally (coincidentally, this is only the interpreter for now :))

And no we can't for some reason, I tried but the FFI implementation would always be called, regardless whether or not there was an intern implementation for this.

Which leads us to the present form, not so cute, but it does the job pretty damn well

**Pull request merged**